src/inference/src/dev/icompiled_model.cpp

```diff
@@ -48,34 +48,47 @@ ov::ICompiledModel::ICompiledModel(const std::shared_ptr<const ov::Model>& model
+            new_param->set_friendly_name(param_name);
+            if (add_operation_names) {
                 OPENVINO_ASSERT(!m_plugin->is_new_api() || leaf_names.find(param_name) == leaf_names.end() ||
                                     param->output(0).get_names().find(param_name) != param->output(0).get_names().end(),
                                 "Model operation names have collisions with tensor names.",
                                 " Please use MO to generate new IR version, it should allow to avoid the issue");
                 leaf_names.insert(param_name);
                 param->output(0).get_tensor().add_names({param_name});
-                m_inputs.emplace_back(
-                    ov::Output<const ov::Node>{param->output(0).get_node(), param->output(0).get_index()});
+                new_param->output(0).get_tensor().add_names({param_name});
             }
-            for (const auto& result : model->get_results()) {
-                auto fake_param = std::make_shared<ov::op::v0::Parameter>(result->get_output_element_type(0),
-                                                                          result->get_output_partial_shape(0));
-                const std::string res_name = ov::op::util::create_ie_output_name(result->input_value(0));
+            new_param->set_element_type(param->get_element_type());
+            new_param->set_layout(param->get_layout());
+            new_param->output(0).get_rt_info() = param->output(0).get_rt_info();
+            new_param->validate_and_infer_types();
+            m_inputs.emplace_back(new_param->output(0));
+        }
+        for (const auto& result : model->get_results()) {
+            auto fake_param = std::make_shared<ov::op::v0::Parameter>(result->get_output_element_type(0),
+                                                                      result->get_output_partial_shape(0));
+            const std::string res_name = ov::op::util::create_ie_output_name(result->input_value(0));
+            fake_param->set_friendly_name(res_name);
+            fake_param->set_element_type(result->get_element_type());
+            fake_param->validate_and_infer_types();
+            auto new_result = result->copy_with_new_inputs({fake_param});
+            new_result->set_friendly_name(result->get_friendly_name());
+            if (add_operation_names) {
                 OPENVINO_ASSERT(!m_plugin->is_new_api() || leaf_names.find(res_name) == leaf_names.end() ||
                                     result->output(0).get_names().find(res_name) != result->output(0).get_names().end(),
                                 "Model operation names have collisions with tensor names.",
                                 " Please use MO to generate new IR version, it should allow to avoid the issue");
                 leaf_names.insert(res_name);
                 result->output(0).get_tensor().add_names({res_name});
```

Comment:
I think we can add names only for new result. and we shouldn't modify original model

Reply:
If not add such names, it will lead to some test failure:  legacy public api + plugin api 2.0 test import model.

`smoke_BehaviorTests/OVCompiledGraphImportExportTest.importExportedIENetwork/targetDevice=CPU_elementType=i8_`

It is caused by name missing when model export.